File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int p = nums[0];
        int count = 1;

        for(int i=1; i<nums.size(); i++){
            if(p != nums[i]){
                if(count == 0){
                    p = nums[i];
                    count = 1;
                }else{
                    count--;;
                }
            }else{
                count++;
            }
        }

        return p;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        curr = 0
        curr_count = 0
        
        for num in nums:
            if curr == num:
                curr_count += 1
            elif curr_count == 0:
                curr = num
                curr_count = 1
            else:
                curr_count -= 1
        
        return curr